backend/matching-service/app/src/matching/matching.gateway.ts

```diff
@@ -31,10 +32,9 @@ export class MatchingGateway {
       }
     );
     const dataJson = await res.json();
+    const safeParsedRoomData = createRoomSchema.safeParse(dataJson);
 
-    // TODO: Properly validate API response schema @Joel
-    if (dataJson.status !== "success" || !dataJson.data) {
-      // TODO: Properly handle error @Joel (instead of crashing the server, it should send an appropriate response to the client)
+    if (!safeParsedRoomData.success) {
```

Comment:
```suggestion
    if (!safeParsedRoomData.success) {
        // This is a server 5xx error 
        throw new Error("Mismatch in expected createRoom schema");
    }
    
    if (safeParsedRoomData.data.status !== "success") {
        throw new Error("Failed to create room");
    }
```

the schema validation step is different from the response success validation step

Reply:
6119054859773dc7ac52299017b295e195a9ed4a